app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -80,6 +91,82 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
     logDir
   }
 
+  /**
+   * Returns the namenode address of the  active nameNode
+   * @param conf The Hadoop configuration
+   * @return The namenode address of the active namenode
+   */
+  def getNamenodeAddress(conf: Configuration): String = {
+
+    // check if the fetcherconf has namenode addresses. There can be multiple addresses and
```

Comment:
this will mean that the elephant conf will override the NAME_SERVICES config even if it exist. If this is a conscious choice of priority order, make that explicit in comments here and in elephant.conf